src/NuGet.Indexing/NuGetSearcherManager.cs

```diff
@@ -240,15 +240,23 @@ protected override NuGetIndexSearcher CreateSearcher(IndexReader reader)
 
                 // There are four flavors of Latest/Listed filter to reflect all the possible combinations.
 
-                var h00 = new LatestListedHandler(includeUnlisted: false, includePrerelease: false);
-                var h01 = new LatestListedHandler(includeUnlisted: false, includePrerelease: true);
-                var h10 = new LatestListedHandler(includeUnlisted: true, includePrerelease: false);
-                var h11 = new LatestListedHandler(includeUnlisted: true, includePrerelease: true);
-
-                indexReaderProcessor.AddHandler(h00);
-                indexReaderProcessor.AddHandler(h01);
-                indexReaderProcessor.AddHandler(h10);
-                indexReaderProcessor.AddHandler(h11);
+                var h000 = new LatestListedHandler(includeUnlisted: false, includePrerelease: false, includeSemver2: false);
```

Comment:
This is a permutation generator. Can be simplified to bitmask of integers from 0 to 2^n - 1. n = 3 today. 😄